tools/protoc-plugin/src/main/java/io/spine/tools/protoc/method/GeneratedMethodScanner.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.protoc.method;
+
+import com.google.common.collect.ImmutableList;
+import io.spine.tools.protoc.CompilerOutput;
+import io.spine.tools.protoc.GenerateMethod;
+import io.spine.tools.protoc.GenerateMethodsConfig;
+import io.spine.tools.protoc.TypeScanner;
+import io.spine.tools.protoc.UuidMethod;
+import io.spine.type.MessageType;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.spine.validate.Validate.isDefault;
+
+/**
+ * Scans the given type for a match upon patterns defined in {@link GenerateMethodsConfig}.
+ */
+final class GeneratedMethodScanner extends TypeScanner<GenerateMethod> {
```

Comment:
This whole thing does not quite read for me. So it's a `TypeScanner` configured with an intention to `GenerateMethod`s. I read it as a scanner telling which messages to generate based on the scan results. But then I see `protected List<GenerateMethod> filePatterns() ` which blows my mind, as it does not fit my understanding at all.